brigade.js

```diff
@@ -0,0 +1,204 @@
+const { events, Job, Group } = require("brigadier");
+
+// **********************************************
+// Globals
+// **********************************************
+
+const projectName = "porter-terraform";
+const projectOrg = "deislabs";
+
+// **********************************************
+// Event Handlers
+// **********************************************
+
+events.on("check_suite:requested", runSuite);
+events.on("check_suite:rerequested", runSuite);
+events.on("check_run:rerequested", runSuite);
+
+events.on("exec", (e, p) => {
+  Group.runAll([
+    build(e, p),
+    xbuild(e, p),
+    test(e, p),
+  ]);
+});
+
+// Although a GH App will trigger 'check_suite:requested' on a push to master event,
+// it will not for a tag push, hence the need for this handler
+events.on("push", (e, p) => {
+  if (e.revision.ref.includes("refs/heads/master") || e.revision.ref.startsWith("refs/tags/")) {
+    publish(e, p).run();
+  }
+});
+
+events.on("publish", (e, p) => {
+  publish(e, p).run();
+});
+
+// **********************************************
+// Actions
+// **********************************************
+
+function build(e, p) {
+  var goBuild = new GoJob(`${projectName}-build`);
+
+  goBuild.tasks.push(
+    "make build"
+  );
+
+  return goBuild;
+}
+
+function xbuild(e, p) {
+  var goBuild = new GoJob(`${projectName}-xbuild`);
+
+  goBuild.tasks.push(
+    "make xbuild-all"
+  );
+
+  return goBuild;
+}
+
+function test(e, p) {
+  var goTest = new GoJob(`${projectName}-test`);
+
+  goTest.tasks.push(
+    "make test-unit"
+  );
+
+  return goTest;
+}
+
+function publish(e, p) {
+  var goPublish = new GoJob(`${projectName}-publish`);
+
+  // TODO: we could/should refactor so that this job shares a mount with the xbuild job above,
+  // to remove the need of re-xbuilding before publishing
+
+  goPublish.env.AZURE_STORAGE_CONNECTION_STRING = p.secrets.azureStorageConnectionString;
+  goPublish.tasks.push(
+    "make xbuild-all publish"
+  )
+
+  return goPublish;
+}
+
+// Here we add GitHub Check Runs, which will run in parallel and report their results independently to GitHub
+function runSuite(e, p) {
+  checkRun(e, p, build, "Build").catch(e  => {console.error(e.toString())});
+  checkRun(e, p, xbuild, "Cross-Platform Build").catch(e  => {console.error(e.toString())});
+  checkRun(e, p, test, "Test").catch(e  => {console.error(e.toString())});
+}
+
+// **********************************************
+// Classes/Helpers
+// **********************************************
+
+// GoJob is a Job with Golang-related prerequisites set up
+class GoJob extends Job {
+  constructor (name) {
+    super(name);
+
+    const gopath = "/go";
+    const localPath = gopath + `/src/github.com/${projectOrg}/${projectName}`;
+
+    // Here using the large-but-useful deis/go-dev image as we have a need for deps
+    // already pre-installed in this image, e.g. helm, az, docker, etc.
+    // TODO: replace with lighter-weight image (Carolyn)
```

Comment:
omg I really do owe you this... 🤦‍♀️

Reply:
:) We'll get to it in our copious free time :)